1-foundations/1-types-and-syntax/foundations-s1.ipynb

```diff
@@ -1,11 +1,1580 @@
+    "print('Third item (index 2):', list_mix[2], type(list_mix[2]))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Access item in list and store in variable\n",
+    "name = list_mix[1]\n",
+    "print('Variable name:', name, type(name))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Accessing items using the index does not modify the list\n",
+    "print('List list_mix:', list_mix, type(list_mix))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Access multiple items in a list:**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Re-create list of ints\n",
+    "list_int = [0,1,2,3,4,5,6,7,8,9]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Get all items between the item with index 0 \n",
+    "# up until but not including the item with index 3\n",
+    "# 0 ≤ index < 3\n",
+    "print(list_int[0:3])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Index 0 is assumed if the fist number is omitted\n",
+    "# So 0:3 is the same as :3\n",
+    "print(list_int[:3])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# 5 ≤ index < 7\n",
+    "print(list_int[5:7])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# 8 ≤ index < infinity\n",
+    "print(list_int[8:])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# (number of items - 3) ≤ index < infinity\n",
+    "print(list_int[-3:])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# (number of items - 7) ≤ index < (number of items - 2)\n",
+    "print(list_int[-7:-2])\n",
+    "# 3 ≤ index < 8\n",
+    "print(list_int[3:8])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Important error message: IndexError**\n",
+    "    \n",
+    "Whenever you see an error where it says \"index out of range\", as in `IndexError: list index out of range`, then it means that you have tried to access an item in the list, using an index that is not used in the list."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# IndexError: list index out of range\n",
+    "print(list_int[10])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Edit a list:**\n",
```

Comment:
suggest adding the methods `pop()` and `remove()` to show how to remove elements by index or value

Reply:
done in: 360840516a909edfe6038351da81f415d8e85b2f